### DISCUSSION TITLE
File size too big

### DISCUSSION THREAD

**Discussion by muriloime** [alpinejs/alpine#745] [4. General]
Hello there, I am very interested in the project, however unpacked size seems to big IMHO ( >700kb, compared to 200kb of react for example). 

Is this correct or am I missing something here ( maybe some tool similar to tailwind's common workflow like to purge unused css like postcss? )? 

Thanks in advance

**Comment by KevinBatdorf** [comment]
How are you bundling it? This entire GitHub repo is only 287kb. 
Straight from the [CDN](https://cdn.jsdelivr.net/gh/alpinejs/alpine/dist/alpine.min.js) is 26kb

**Comment by muriloime** [comment]
Hey thanks @KevinBatdorf . I am taking npm "Unpacked size" for granted at [alpinejs npm](https://www.npmjs.com/package/alpinejs). it is currently showing 774 kB. I wonder why such a big difference with the numbers you gave

**Reply by ryangjchandler** [reply]
You might want to use this tool to find the true size of files on disk: https://arve0.github.io/npm-download-size/#alpinejs

It's worth noting that this includes all of the files NPM downloads when you run `npm install alpinejs`.

**Reply by muriloime** [reply]
These numbers look much better. I was finding contradictory the size with the name of the library :laughing:  . Will give it a try . Many thanks !

**Reply by HugoDF** [reply]
Here's another link for bundle size https://bundlephobia.com/result?p=alpinejs

I opened a PR once to reduce unpacked size (we're shipping a lot of stuff that's not necessary in the npm package)

**Reply by KevinBatdorf** [reply]
Does NPM calculate the entire repo plus all the dependencies? 

It might be worth getting that to a realistic number as it could be turning people away from the project.

**Reply by HugoDF** [reply]
> Does NPM calculate the entire repo plus all the dependencies?

Npm uses the packed/unpacked size of the package, which at the moment is the whole repository minus the `node_modules` folder. That means both `dist` builds, but also all the source, all the HTML files and all the tests/docs etc.

See this https://github.com/alpinejs/alpine/issues/244
